### PR TITLE
Option interactive

### PR DESCRIPTION
**Pull request by rurikoaraki**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Adding interactivity state & tokens, and default styling for hovered, pressed, and disabled states.

The changes are:

- Add componentRef (which in retrospect, wasn't necessary, but it will be in the future)
- Add useAsPressable to useOption, and then return pressable props and state.
- Add interaction states and tokens, and add default tokens for those states.
- Include the new states in the lookup function for useSlots, and then merge the output of useOption into the final props

### Verification

option_interaction

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Overall question: any reason this is a new component rather than reusing menu item?

**Comment by rurikoaraki** [comment]
> Overall question: any reason this is a new component rather than reusing menu item?

Dropdown is semantically different from Menu, so that's the reason why there's new components instead of reusing Menu to build Dropdown. In the future Option will have different accessibility settings, etc.

There's probably opportunity to refactor code to share behavior logic between the two, though

File: packages/experimental/Dropdown/src/Option/useOption.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
 
-export const useOption = (_props: OptionProps): OptionState => {
-  return {
-    props: {},
-    state: {},
-  };
+export const useOption = (props: OptionProps): OptionState => {
+  const pressable = useAsPressable(props);
+
+  return pressable;
```
Comment (Saadnajmi) [inline packages/experimental/Dropdown/src/Option/useOption.ts:7]: (Feel free to reject this suggestion)
Could we try using Pressable from `react-native` along with `usePressableState` for new components? Super Ok if you'd rather leave that for later.

File: packages/experimental/Dropdown/src/Option/Option.styling.ts [hunk 2]
```diff
@@ -2,21 +2,25 @@ import { borderStyles, buildProps, fontStyles, layoutStyles, Theme, UseStylingOp
 import { optionName, OptionProps, OptionSlotProps, OptionTokens } from './Option.types';
 import { defaultOptionTokens } from './OptionTokens';
 
+export const optionStates: (keyof OptionTokens)[] = ['hovered', 'focused', 'pressed', 'disabled'];
```
Comment (Saadnajmi) [inline packages/experimental/Dropdown/src/Option/Option.styling.ts:5]: It feels like we might or should have a type for these 4 states in particular we can spread? Future fodder if not.. I don't know that would make code cleaner..

**Review by Saadnajmi — APPROVED** [review]

**Review by samuelfreiberg — APPROVED** [review]

File: packages/experimental/Dropdown/src/Option/Option.styling.ts (hunk 2 above)
Comment (rurikoaraki) [reply packages/experimental/Dropdown/src/Option/Option.styling.ts:5]: I don't know how well that would play with typing. We can try it, I guess a template might make it feasible

File: packages/experimental/Dropdown/src/Option/useOption.ts (hunk 1 above)
Comment (rurikoaraki) [reply packages/experimental/Dropdown/src/Option/useOption.ts:7]: Is Pressable ready for use?

Comment (Saadnajmi) [reply packages/experimental/Dropdown/src/Option/useOption.ts:7]: It should be. The only think lacking from when I last worked on it was types in the DefinitelyTyped repo.

File: packages/experimental/Dropdown/src/Option/Option.styling.ts (hunk 2 above)
Comment (Saadnajmi) [reply packages/experimental/Dropdown/src/Option/Option.styling.ts:5]: I think I was confusing old and new framework, and different types. `useAsPressable.types.ts` has "I<Focus | Press | Hover>State` types. Nevermind..

**Review by gulnazsayed — APPROVED** [review]

File: packages/experimental/Dropdown/src/Option/useOption.ts (hunk 1 above)
Comment (rurikoaraki) [reply packages/experimental/Dropdown/src/Option/useOption.ts:7]: I think I'll commit this for now so it doesn't get too out of sync from main, and work on the Pressable separately.

**Pull request merged**